shell/shared/grain.js

```diff
@@ -559,11 +666,32 @@ if (Meteor.isClient) {
     displayToken: function() {
       return !this.revoked && !this.expiresIfUnused && !this.parentToken;
     },
+    existingTokens: function () {
+      var current = getActiveGrain(globalGrains.get());
+      return current && ApiTokens.find({grainId: current.grainId(), userId: Meteor.userId(),
+                                        forSharing: {$ne: true},
+                                        $or: [{owner: {webkey: null}},
+                                              {owner: {$exists: false}}],
+                                        expiresIfUnused: null});
+    },
+    generatedApiToken: function () {
+      var current = getActiveGrain(globalGrains.get());
+      return current && current.generatedApiToken();
+    },
+    generatedApiTokenPending: function() {
+      var current = getActiveGrain(globalGrains.get());
+      return (current && current.generatedApiToken()) == "pending";
+    },
+    viewInfo: function () {
```

Comment:
The shareWithOthers template needs a helper like this too, so that the role selection element shows up there.